aw-server/tests/api.rs

```diff
@@ -334,4 +334,110 @@ mod api_tests {
+        let server = setup_testserver();
+        let client = rocket::local::Client::new(server).expect("valid instance");
+
+        // Test getting not found (getting nonexistent key)
+        let res = client.get("/api/0/settings/thisisaverylongkthisisaverylongkthisisaverylongkthisisaverylongkthisisaverylongkthisisaverylongkthisisaverylongkthisisaverylongk")
+            .dispatch();
+        assert_eq!(res.status(), rocket::http::Status::BadRequest);
+    }
+
+    #[test]
+    fn test_creating_value() {
+        let server = setup_testserver();
+        let client = rocket::local::Client::new(server).expect("valid instance");
+
+        // Test value creation
+        let response_status = create_value_request(&client, "test_key");
+        assert_eq!(response_status, rocket::http::Status::Created);
+
+    }
+
+    #[test]
+    fn test_getting_not_found_value() {
+        let server = setup_testserver();
+        let client = rocket::local::Client::new(server).expect("valid instance");
+
+        // Test getting not found (getting nonexistent key)
+        let res = client.get("/api/0/settings/non_existent_key")
+        .dispatch();
+        assert_eq!(res.status(), rocket::http::Status::NotFound);
+    }
+    /// TODO: Add a test for the settings list here
+    #[test]
+    fn settings_list_get() {
+        let server = setup_testserver();
+        let client = rocket::local::Client::new(server).expect("valid instance");
+
+        let response1_status = create_value_request(&client, "test_key");
+        assert_eq!(response1_status, rocket::http::Status::Created);
+        let response2_status = create_value_request(&client, "test_key");
+        assert_eq!(response2_status, rocket::http::Status::Created);
+
+        let mut res = client.get("/api/0/settings/").dispatch();
+        
+        assert_eq!(res.status(), rocket::http::Status::Ok);
+        assert_eq!(res.body_string().unwrap(), r#""#);
+        
+    }
+
+    #[test]
+    fn test_getting_setting() {
+        let server = setup_testserver();
+        let client = rocket::local::Client::new(server).expect("valid instance");
+
+        let response_status = create_value_request(&client, "test_key");
+        assert_eq!(response_status, rocket::http::Status::Created);
+
+        // Test getting
+        let mut res = client.get("/api/0/settings/test_key").dispatch();
+        assert_eq!(res.status(), rocket::http::Status::Ok);
+        assert_eq!(res.body_string().unwrap(), r#"test_value"#);
+    }
+
+    #[test]
+    fn test_updating_setting() {
+        let server = setup_testserver();
+        let client = rocket::local::Client::new(server).expect("valid instance");
+
+        let response_status = create_value_request(&client, "test_key");
+        assert_eq!(response_status, rocket::http::Status::Created);
+
+        let res = client.post("/api/0/settings/test_key")
+            .header(ContentType::JSON)
+            .body(r#""changed_test_value""#)
```

Comment:
If you modify create_value_request to also take in the value you could re-use it here.

Reply:
Done in abcfc49